signer/xpi/xpi.go

```diff
@@ -315,9 +326,28 @@ func (o *Options) Algorithms() (algs []*cose.Algorithm, err error) {
 	return
 }
 
+// PK7Digest validates and return an ASN OID for a PKCS7 digest
+// algorithm defaulting to SHA1
+func (o *Options) PK7Digest() (asn1.ObjectIdentifier) {
+	if o == nil {
+		return pkcs7.OIDDigestAlgorithmSHA1
+	}
+	switch strings.ToUpper(o.PKCS7Digest) {
+	case "SHA256":
+		return pkcs7.OIDDigestAlgorithmSHA256
+	case "SHA1":
+		return pkcs7.OIDDigestAlgorithmSHA1
+	default:
+		return pkcs7.OIDDigestAlgorithmSHA1
```

Comment:
maybe default should error out instead of falling back to sha1 if the algorithm is unknown?